src/ExportModule.ts

```diff
@@ -32,10 +32,19 @@ export function exportModule(libName: string, dataTypes: CalculatorTypes.DataTyp
 
   dataTypes.forEach((object, i) => {
     const stateName = `${object.dataType}_${i}`;
+
     if (object.dataType !== null && factory(object.dataType, stateName) !== null) {
-      const StateClass = factory(object.dataType, stateName);
+      const stateClass = factory(object.dataType, stateName);
+      if (object.valueSet !== undefined && stateClass && states.doesHaveCodes(stateClass)) {
+        stateClass['codes'].push({
+          system: '',
+          code: '',
+          display: '',
+          value_set: object.valueSet
+        });
+      }
 
-      StateClass && (moduleJSON.states[stateName] = StateClass.toJSON());
+      stateClass && (moduleJSON.states[stateName] = stateClass.toJSON());
```

Comment:
Actually I see what it does. If the stateClass was not created for some reason it would shortcut out and not try to add the undefined state to the module. I would rather see an `if` statement that checks that.